ui/src/app/views/account-creation/component.tsx

```diff
@@ -0,0 +1,312 @@
+
+import {
+  TooltipTrigger
+} from 'app/components/popups';
+
+import {Button} from 'app/components/buttons';
+import { FormSection } from 'app/components/forms';
+import {BoldHeader} from 'app/components/headers';
+
+function isBlank(s: string) {
+  return (!s || /^\s*$/.test(s));
+}
+
+export interface AccountCreationProps {
+  invitationKey: string;
+  setProfile: Function;
+  onAccountCreation: Function;
+}
+
+export interface AccountCreationState {
+  profile: Profile;
+  usernameCheckInProgress: boolean;
+  usernameConflictError: boolean;
+  creatingAccount: boolean;
+  showAllFieldsRequiredError: boolean;
+}
+
+export class AccountCreationReact extends
+    React.Component<AccountCreationProps, AccountCreationState> {
+  usernameCheckTimeout: NodeJS.Timer;
+  accountCreated = false;
+
+  constructor(props: AccountCreationProps) {
+    super(props);
+    this.state = {
+      profile: {
+        // Note: We abuse the "username" field here by omitting "@domain.org". After
+        // profile creation, this field is populated with the full email address.
+        username: '',
+        dataAccessLevel: DataAccessLevel.Unregistered,
+        givenName: '',
+        familyName: '',
+        contactEmail: '',
+        currentPosition: '',
+        organization: '',
+        areaOfResearch: '',
+      },
+      usernameCheckInProgress: false,
+      usernameConflictError: false,
+      creatingAccount: false,
+      showAllFieldsRequiredError: false
+    };
+    this.createAccount = this.createAccount.bind(this);
+    this.usernameValid = this.usernameValid.bind(this);
+    this.usernameNotValid = this.usernameNotValid.bind(this);
+  }
+
+
+  createAccount(): void {
+    const {invitationKey, setProfile, onAccountCreation} = this.props;
+    if (this.state.usernameConflictError || this.usernameInvalidError()) {
+      return;
+    }
+    this.setState({showAllFieldsRequiredError: false});
+    const requiredFields =
+      [this.state.profile.givenName, this.state.profile.familyName,
+        this.state.profile.username, this.state.profile.contactEmail,
+        this.state.profile.currentPosition, this.state.profile.organization,
+        this.state.profile.areaOfResearch];
+    if (requiredFields.some(isBlank)) {
+      this.setState({showAllFieldsRequiredError: true});
+      return;
+    } else if (this.isUsernameValidationError()) {
+      return;
+    }
+    const request: CreateAccountRequest = {
+      profile: this.state.profile,
+      invitationKey: invitationKey
+    };
+    this.setState({creatingAccount: true});
+    const args: FetchArgs = ProfileApiFetchParamCreator().createAccount(request);
+    fetch(fullUrl(args.url), args.options)
+      .then(handleErrors)
+      .then((response) => response.json)
+      .then((profile) => {
+          this.setState({profile: profile, creatingAccount: false});
+          this.accountCreated = true;
+          setProfile(profile);
+          onAccountCreation();
```

Comment:
Since these two callbacks are called together, they could be combined. E.g. we could just pass the profile to `onAccountCreation`, and the parent could save it and change the step at the same time. (Assuming the parent really needs the profile, see other comment)